go/vt/vttablet/tabletserver/state_manager.go

```diff
@@ -850,3 +854,10 @@ func (sm *stateManager) IsServingString() string {
 func (sm *stateManager) SetUnhealthyThreshold(v time.Duration) {
 	sm.unhealthyThreshold.Store(v.Nanoseconds())
 }
+
+// markClusterAction marks whether a cluster action is in progress or not for all the query details.
+func (sm *stateManager) markClusterAction(ca ClusterActionState) {
+	sm.statefulql.SetClusterAction(ca)
+	sm.statelessql.SetClusterAction(ca)
+	sm.olapql.SetClusterAction(ca)
+}
```

Comment:
over here we should ignore `ClusterActionNoQueries` when the state is `ClusterActionNotInProgress`

Reply:
I can see the check is in `func (ql *QueryList) SetClusterAction(ca ClusterActionState)`